validator/tasks/scheduler.py

```diff
@@ -24,59 +22,69 @@
 
 
 async def _get_models(keypair: Keypair) -> AsyncGenerator[str, None]:
-    response: list[dict[str, Any]] = await call_content_service(csts.GET_RANDOM_MODELS_ENDPOINT, keypair)
-    models = [ModelData(**model) for model in response]
-    random.shuffle(models)
-    for m in models:
-        yield m.model_id
+    response = await call_content_service(cst.GET_RANDOM_MODELS_ENDPOINT, keypair)
+    if not isinstance(response, list):
+        raise TypeError("Expected a list of responses from GET_ALL_MODELS_ENDPOINT")
+    models: list[dict[str, Any]] = response
+    TEMP_MODEL_FAMILIES_ACCEPTED = ["qwen", "llama", "falcon", "mistral", "gemma", "gemini", "phi"]
+    model_ids = [
+        model.get(cst.GET_ALL_MODELS_ID, "")
+        for model in models
+        if any(family in model.get(cst.GET_ALL_MODELS_ID, "").lower() for family in TEMP_MODEL_FAMILIES_ACCEPTED)
+    ]
+    random.shuffle(model_ids)
+    for model_id in model_ids:
+        yield model_id
 
 
 async def _get_datasets(keypair: Keypair) -> AsyncGenerator[str, None]:
-    response: list[dict[str, Any]] = await call_content_service(csts.GET_RANDOM_DATASETS_ENDPOINT, keypair)
-    datasets = [DatasetData(**ds) for ds in response]
-    random.shuffle(datasets)
-    for ds in datasets:
-        yield ds.dataset_id
-
-
-async def _get_the_columns_for_dataset(dataset_id: str, keypair: Keypair) -> DatasetRequest:
-    url = csts.GET_COLUMNS_FOR_DATASET_ENDPOINT.replace(
-        "{dataset}", dataset_id)
+    response = await call_content_service(cst.GET_RANDOM_DATASETS_ENDPOINT, keypair)
+    if not isinstance(response, list):
```

Comment:
AsyncGenerator[str, None]:
    response: list[dict[str, Any]] = await call_content_service(csts.GET_RANDOM_DATASETS_ENDPOINT, keypair)
    datasets = [DatasetData(**ds) for ds in response]
    random.shuffle(datasets)
    for ds in datasets:
        yield ds.dataset_id
        
        Cleaner?